agnostic-lite/src/smol.rs

```diff
@@ -51,12 +51,12 @@ impl AsyncSpawner for SmolSpawner {
 }
 
 impl AsyncLocalSpawner for SmolSpawner {
-  type JoinHandle<O>
+  type SmolJoinHandle<O>
```

Comment:
The associated type name `JoinHandle` is changed to `SmolJoinHandle`, this should not be changed